stl/inc/ranges

```diff
@@ -3603,14 +3605,28 @@ namespace ranges {
         };
 
         template <bool _Const>
-        class _Iterator
+        class _Iterator_base
+            : public _Category_base<_Maybe_const<_Const, _Vw>, _InnerRng<_Const>, is_reference_v<_InnerRng<_Const>>> {};
+
+        template <bool _Const>
+            requires forward_range<_Vw>

```

Comment:
Should this say `requires forward_range<_Maybe_const<_Const, _Vw>>` instead? I see that \[range.join.iterator\]'s `OuterIter outer_ = OuterIter();` is being patched to say `present only if Base models forward_range` where `Base` is `maybe-const<Const, V>`.